pkg/glusterfs/controllerserver.go

```diff
@@ -133,17 +124,23 @@ func validateThinArbiter(req *csi.CreateVolumeRequest) error {
 	return nil
 }
 
-func addThinArbiter(req *api.VolCreateReq, thinArbiter string) error {
+func addThinArbiter(volReq *api.VolCreateReq, req *csi.CreateVolumeRequest) error {
+
+	if err := validateThinArbiter(req); err != nil {
+		return err
+	}
 
+	thinArbiter := req.Parameters["arbiterPath"]
 	s := strings.Split(thinArbiter, ":")
 	if len(s) != 2 && len(s) != 3 {
 		return fmt.Errorf("thin arbiter brick must be of the form <host>:<brick> or <host>:<brick>:<port>")
 	}
 
-	req.Options = map[string]string{
+	volReq.ReplicaCount = 2
```

Comment:
Why this is hardcoded to "2" ? it should pick it up from a constant

Reply:
currently gd2 supports replica 2 and arbiter 1